### ISSUE TITLE
The problem of memory leak

### ISSUE THREAD

**Issue by renyuanzhe** [oppo-us-research/USST#1]
The cpu ram usage increases  every epoch and every batch in one epoch ,probably caused by the ProbTranV3 module. Is someone encounter this issue?

**Comment by Cogito2012** [comment]
Hi @renyuanzhe, thank you for your interest!
I haven't met this issue before, but I will have a test to monitor the CPU usage these days. I'll let you know ASAP.

**Comment by renyuanzhe** [comment]
hello, im sorry to say that i misunderstood the ProbTranV3 module, which is not the cause of memory leak. 
Interestingly，if i use the cpu to train the model, the memory leak will not appear. If i use gpu to train the model ,the memory usage will increase every batch.
Hoping for your help , thankyou.

**Comment by Cogito2012** [comment]
@renyuanzhe Thanks for providing this information. 
I'm curious about how you monitor the memory usage. I tried to rerun the training script with one GPU (id=0) and 8 multi-thread workers, and use the `ps` command to monitor the CPU & memory usage. It did not show an increasing usage in training, even after many epochs.

In one terminal, run the training in a backend screen session:
```shell
screen -S rerun_usst
cd exp
nohup bash train.sh 0 8 usst_vit_3d >train_egopat3d.log 2>&1 &
```

In another terminal, monitoring the CPU and memory usage (every 5 seconds):
```shell
watch -n 5 'ps -u baowenta -o pid,%cpu,%mem,cmd | grep -E "python"'
```
![cpu_usage](https://github.com/oppo-us-research/USST/assets/13028018/cccd1e68-146a-4ba9-8fdb-7e54c2b06209)

**Comment by renyuanzhe** [comment]
@Cogito2012 
Deer writer, I run your project on windows10 platform, using the same requirements.txt. 

Additionally, I use the memory_profiler to monitor at which step the menmory increases
`python -m memory_profiler trainval_h2o.py --num_workers 0  --config config/h2o/usst_res18_3d.yml  --tag h2o/usst_res18_3d`

and find that probably it is at the state transition module in TransformerSSM.py
`future_pred, pred_dict = self.transition(mask_o, vis_obs=vis_feat, traj_obs=traj_feat) `

The Screen recording of the training process and the memory_profiler output are sent to your mailbox.

**Comment by Cogito2012** [comment]
@renyuanzhe I received your recording, thanks very much.
I tested with the same command as yours on my GPU server. It indeed showed an increasing usage of RAM, but it eventually saturated at around 11 GB (`rss` output of `ps` command.) in 3~4 epochs and will not cause any error.

The only factor that I could imagine is the tensorboard writer, which cumulatively adds loss values into the writer object (in CPU memory). But this impact should be very minor.

I was wondering if you will encounter any issues if disregard the memory increase.

**Comment by renyuanzhe** [comment]
@Cogito2012 
Thank you for your reply. I use a subset of the dataset to train the model, so that the memory usage will not exceed the total sum of my device. Just as you said, the memory usage eventually saturated after 3 or 4 epochs. I wonder whether it is the line 'loss += losses['total_loss']' in runtime.py that leads to this situation. If I replace it with the 'loss += losses['total_loss'].item()', the memory usage will not increase forever.

**Comment by Cogito2012** [comment]
@renyuanzhe Yeah, it could be that reason. Thank you!
